Utilities/Thread.cpp

```diff
@@ -1268,7 +1268,7 @@ bool handle_access_violation(u32 addr, bool is_writing, x64_context* context)
 					// Now, we notify the game that a page fault occurred so it can rectify it.
 					// Note, for data3, were the memory readable AND we got a page fault, it must be due to a write violation since reads are allowed.
 					be_t<u64> data1 = addr;
-					be_t<u64> data2 = (SYS_MEMORY_PAGE_FAULT_TYPE_PPU_THREAD << 32) + cpu->id; // TODO: fix hack for now that assumes PPU thread always.
+					be_t<u64> data2 = ((be_t<u64>)SYS_MEMORY_PAGE_FAULT_TYPE_PPU_THREAD << 32) + cpu->id; // TODO: fix hack for now that assumes PPU thread always.
```

Comment:
while at it, fix the type of `data1` to `data3`, it should be u64. (same for SYS_MEMORY_PAGE_FAULT_TYPE_PPU_THREAD)

Reply:
done